File: src/main/java/net/hectus/neobb/turn/default_game/other/TTimeLimit.java
package net.hectus.neobb.turn.default_game.other;

import com.marcpg.libpg.data.time.Time;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.Turn;
import net.hectus.neobb.turn.default_game.attributes.clazz.NeutralClazz;
import net.hectus.neobb.turn.default_game.attributes.usage.OtherUsage;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

public class TTimeLimit extends Turn<Time> implements OtherUsage<Time>, NeutralClazz {
    public TTimeLimit(NeoPlayer player) { super(null, null, player); }
    public TTimeLimit(Time data, NeoPlayer player) { super(data, player.player.getLocation(), player); }

    @Override
    public ItemStack item() {
        return new ItemStack(Material.CLOCK);
    }

    @Override
    public int cost() {
        return 0;
    }

    @Override
    public Time getValue() {
        return data;
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/TTimeLimit.kt
package net.hectus.neobb.modes.turn.default_game

import com.marcpg.libpg.data.time.Time
import com.marcpg.libpg.util.component
import net.hectus.neobb.event.TurnEvent
import net.hectus.neobb.modes.turn.TurnExec
import net.hectus.neobb.modes.turn.default_game.attribute.TurnClazz
import net.hectus.neobb.util.Colors
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

object TTimeLimit : OtherTurn<Time>("time_limit") {
    override val mode: String = "any"
    override val event: TurnEvent = TurnEvent.NONE
    override val clazz: TurnClazz = TurnClazz.NEUTRAL
    override val mainItem: ItemStack = ItemStack.of(Material.CLOCK)

    override fun apply(exec: TurnExec<Time>) {
        exec.player.sendMessage(exec.player.locale().component("gameplay.info.too-slow.player", exec.data.toString(), color = Colors.NEGATIVE))
        exec.player.opponents(false).forEach { p ->
            p.sendMessage(p.locale().component("gameplay.info.too-slow." + (if (p == exec.player.targetPlayer()) "next" else "others"), exec.player.targetPlayer().name(), color = Colors.NEUTRAL))
        }
    }
}
